manifests/config.pp

```diff
@@ -45,15 +49,16 @@
     $_dirs = [
       "${keycloak::install_base}/domain/servers",
       "${keycloak::install_base}/domain/servers/${keycloak::server_name}",
-      "${keycloak::install_base}/domain/servers/${keycloak::server_name}/configuration",
+      $_server_conf_dir
     ]
 
-    file { $_dirs:
-      ensure => 'directory',
-      owner  => $keycloak::user,
-      group  => $keycloak::group,
-      mode   => '0755',
-    }
+  }
+
+  file { $_dirs:
```

Comment:
Per other comment, it's better to only use this `$_dirs` resource where actually needed which is when domain mode is enabled.  The standalone it's not needed since the directory is already present during install, Puppet just manages the permissions.